test/submit_05_reorg_test.go

```diff
@@ -0,0 +1,147 @@
+//go:build e2e
+
+package test
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/bitcoin-sv/arc/internal/node_client"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReorg(t *testing.T) {
+	address, privateKey := node_client.FundNewWallet(t, bitcoind)
+
+	utxos := node_client.GetUtxos(t, bitcoind, address)
+	require.True(t, len(utxos) > 0, "No UTXOs available for the address")
+
+	tx1, err := node_client.CreateTx(privateKey, address, utxos[0])
+	require.NoError(t, err)
+
+	// submit tx1
+	rawTx, err := tx1.EFHex()
+	require.NoError(t, err)
+	resp := postRequest[TransactionResponse](t, arcEndpointV1Tx, createPayload(t, TransactionRequest{RawTx: rawTx}), map[string]string{"X-WaitFor": StatusSeenOnNetwork}, http.StatusOK)
+	require.Equal(t, StatusSeenOnNetwork, resp.TxStatus)
+
+	// mine tx1
+	invHash := node_client.Generate(t, bitcoind, 1)
+
+	// verify tx1 = MINED
+	statusURL := fmt.Sprintf("%s/%s", arcEndpointV1Tx, tx1.TxID())
+	statusResp := getRequest[TransactionResponse](t, statusURL)
+	require.Equal(t, StatusMined, statusResp.TxStatus)
+	require.Equal(t, invHash, *statusResp.BlockHash)
+
+	// get new UTXO for tx2
+	node_client.SendToAddress(t, bitcoind, address, float64(0.002))
+	utxos = node_client.GetUtxos(t, bitcoind, address)
+	require.True(t, len(utxos) > 0, "No UTXOs available for the address")
+
+	// make sure to pick the correct UTXO
+	var utxo node_client.UnspentOutput
+	for _, u := range utxos {
+		if u.Amount == float64(0.002) {
+			utxo = u
+		}
+	}
+
+	tx2, err := node_client.CreateTx(privateKey, address, utxo)
+	require.NoError(t, err)
+
+	// submit tx2
+	rawTx, err = tx2.EFHex()
+	require.NoError(t, err)
+	resp = postRequest[TransactionResponse](t, arcEndpointV1Tx, createPayload(t, TransactionRequest{RawTx: rawTx}), map[string]string{"X-WaitFor": StatusSeenOnNetwork}, http.StatusOK)
+	require.Equal(t, StatusSeenOnNetwork, resp.TxStatus)
+
+	// mine tx2
+	tx2BlockHash := node_client.Generate(t, bitcoind, 1)
+
+	// verify tx2 = MINED
+	statusURL = fmt.Sprintf("%s/%s", arcEndpointV1Tx, tx2.TxID())
+	statusResp = getRequest[TransactionResponse](t, statusURL)
+	require.Equal(t, StatusMined, statusResp.TxStatus)
+	require.Equal(t, tx2BlockHash, *statusResp.BlockHash)
+
+	// invalidate the chain with tx1 and tx2
+	call(t, "invalidateblock", []interface{}{invHash})
+
+	// prepare txStale
+	node_client.SendToAddress(t, bitcoind, address, float64(0.003))
+
+	utxos = node_client.GetUtxos(t, bitcoind, address)
+	require.True(t, len(utxos) > 0, "No UTXOs available for the address")
+
+	// make sure to pick the correct UTXO
+	for _, u := range utxos {
+		if u.Amount == float64(0.003) {
+			utxo = u
```

Comment:
I think you could also find this utxo by the ID
```go
	txID := node_client.SendToAddress(t, bitcoind, address, float64(0.002))
	utxos = node_client.GetUtxos(t, bitcoind, address)
	require.True(t, len(utxos) > 0, "No UTXOs available for the address")

	// make sure to pick the correct UTXO
	var utxo node_client.UnspentOutput
	for _, u := range utxos {
		if u.Txid == txID {
			utxo = u
		}
	}
```

Reply:
Changed